(macro-hygiene) ramhorn derive codegen clashes with syn::Result

Hi @maciejhirsz 

Thank you for this library!

While using `ramhorns` in my project: I have a use case where I need to use both `syn` and `ramhorn` together.

Following is the minimal reproducible example:

```rust

use syn::Result;

use ramhorns::{Content};

#[derive(Content)]
struct Post<'a> {
    title: &'a str
}

```

But, On running `cargo build`, I get the following error:
```bash
error[E0107]: wrong number of type arguments: expected 1, found 2
  --> src/main.rs:53:10
   |
53 | #[derive(Content)]
   |          ^^^^^^^ unexpected type argument
   |
   = note: this error originates in a derive macro (in Nightly builds, run with -Z macro-backtrace for more info)

error: aborting due to previous error; 7 warnings emitted

For more information about this error, try `rustc --explain E0107`.
```

From what i understand, The code generated by `ramhorn-derive` generates `Result<...>` on `Content`'s traits methods and that clashes with syn::Result. And as you can see the error message is also misleading.

What's the best way to solve this? I could alias my import of syn, but consider a larger project which imports ramhorn and derive and also has a `syn::Result` imported already, they might not know where to look for this error. As my example was small, I was able remove code incrementally and figured this out.

--- a/ramhorns-derive/src/lib.rs
+++ b/ramhorns-derive/src/lib.rs
@@ -231,7 +231,7 @@ pub fn content_derive(input: TokenStream) -> TokenStream {
             }
 
             #[inline]
-            fn render_section<C, E>(&self, section: ramhorns::Section<C>, encoder: &mut E) -> Result<(), E::Error>
+            fn render_section<C, E>(&self, section: ramhorns::Section<C>, encoder: &mut E) -> std::result::Result<(), E::Error>
             where
                 C: ramhorns::traits::ContentSequence,
                 E: ramhorns::encoding::Encoder,
@@ -240,7 +240,7 @@ pub fn content_derive(input: TokenStream) -> TokenStream {
             }
 
             #[inline]
-            fn render_field_escaped<E>(&self, hash: u64, name: &str, encoder: &mut E) -> Result<bool, E::Error>
+            fn render_field_escaped<E>(&self, hash: u64, name: &str, encoder: &mut E) -> std::result::Result<bool, E::Error>
             where
                 E: ramhorns::encoding::Encoder,
             {
@@ -254,7 +254,7 @@ pub fn content_derive(input: TokenStream) -> TokenStream {
             }
 
             #[inline]
-            fn render_field_unescaped<E>(&self, hash: u64, name: &str, encoder: &mut E) -> Result<bool, E::Error>
+            fn render_field_unescaped<E>(&self, hash: u64, name: &str, encoder: &mut E) -> std::result::Result<bool, E::Error>
             where
                 E: ramhorns::encoding::Encoder,
             {
@@ -267,7 +267,7 @@ pub fn content_derive(input: TokenStream) -> TokenStream {
                 }
             }
 
-            fn render_field_section<P, E>(&self, hash: u64, name: &str, section: ramhorns::Section<P>, encoder: &mut E) -> Result<bool, E::Error>
+            fn render_field_section<P, E>(&self, hash: u64, name: &str, section: ramhorns::Section<P>, encoder: &mut E) -> std::result::Result<bool, E::Error>
             where
                 P: ramhorns::traits::ContentSequence,
                 E: ramhorns::encoding::Encoder,
@@ -281,7 +281,7 @@ pub fn content_derive(input: TokenStream) -> TokenStream {
                 }
             }
 
-            fn render_field_inverse<P, E>(&self, hash: u64, name: &str, section: ramhorns::Section<P>, encoder: &mut E) -> Result<bool, E::Error>
+            fn render_field_inverse<P, E>(&self, hash: u64, name: &str, section: ramhorns::Section<P>, encoder: &mut E) -> std::result::Result<bool, E::Error>
             where
                 P: ramhorns::traits::ContentSequence,
                 E: ramhorns::encoding::Encoder,

